Issue 129 carriage returns

syavorsky opened the pull request:
proposed fix for #129
- keep CR as is
- add `crlf('LF' | 'CRLF')` transformation to be used with `stringify()`

brettz9 commented:
Thanks for the fix! It appears to be preserving the necessary carriage returns but I need more time to integrate it in our code base and see if the issues remaining for us are just on our end (as I think they may be). Hopefully will get to it within the next week.

syavorsky commented:
Yeah, good then. I will publish the non-beta once whenever you confirm there are no observed side effects. Thanks

brettz9 commented:
It looks like our one remaining issue is related to our own copy of https://github.com/syavorsky/comment-parser/blob/master/src/transforms/align.ts . While we can fix it ourselves, in case you were interested in, um, "aligning" it on your end, the specific line is https://github.com/syavorsky/comment-parser/blob/master/src/transforms/align.ts#L61 :

```js
      tokens.postDelimiter = tokens.description === '' ? '' : ' ';
```

I've fixed this for our case by changing to the following:

```js
      tokens.postDelimiter = tokens.description === '' ?
        (tokens.postDelimiter === '\r' ? '\r' : '') :
        ' ';
```

Thus, we don't drop single carriage returns.

I imagine your code at https://github.com/syavorsky/comment-parser/blob/master/src/transforms/align.ts#L91 :

```js
    tokens.postDelimiter = nothingAfter.delim ? '' : space(spacings.postDelimiter);
```

...might also need this treatment:

```js
    tokens.postDelimiter = nothingAfter.delim ? (tokens.postDelimiter === '\r' ? '\r' : '') : space(spacings.postDelimiter);
```

syavorsky commented:
Good point, will make this change too

brettz9 commented:
Btw, @syavorsky whether you can get to the align issue mentioned in [this comment](https://github.com/syavorsky/comment-parser/pull/132#issuecomment-876814403) now or not, do you think you could release a new beta which includes the new ESM changes already merged as part of #130?

syavorsky commented:
@brettz9 sure, done
```
+ comment-parser@1.1.6-beta.1
```

syavorsky commented:
@brettz9 I finally got to look into `transform/align` issues. I have changed the approach and split CRs into separate `.lineEnd` token. This way it concerns only `strngfy()` when all tokens are being put back together. Added the test for the aligning case.

when you get a chance please check out `1.1.6-beta.3`

brettz9 commented:
Seems ok so far--Plan to check it out a little more, but for now, just to let you know if you want to add `lineEnd: 1` to `seedTokens`.

brettz9 commented:
Yeah, seems good from what I can tell though I think I could use that `seedTokens` update (adding `lineEnd: ''` to it) if you wouldn't mind adding that and releasing.

brettz9 commented:
Btw, you might also want to update `engines` to `^12.20 || ^14.14.0 || ^16`, as the `exports` field seems to have only been properly understood after those versions (14.13.0 was supposedly ok with them, but it had another bug requiring 14.14.0 in our tests)...  The version `>=12.0.0` is probably ok for those just using the main `comment-parser` import/require and maybe even `comment-parser/util`, but the other ranges are needed for paths like `comment-parser/parser/tokenizers/description`. (And in case you missed it from the previous PR, `/lib/` (or `/esm/`) are no longer required (or allowed) for subpaths as `exports` auto-adds this now.)

syavorsky commented on src/util.ts:
```diff
@@ -51,6 +62,7 @@ export function seedTokens(tokens: Partial<Tokens> = {}): Tokens {
     postType: '',
     description: '',
     end: '',
+    lineEnd: '',
```
@brettz9 can you elaborate on what you meant? Do I miss the default `lineEnd: ''` somewhere else?
> Yeah, seems good from what I can tell though I think I could use that seedTokens update (adding lineEnd: '' to it) if you wouldn't mind adding that and releasing.

brettz9 replied:
Oh, very strange--sorry, I must have been searching through my copy on `master`.

syavorsky commented:
sure, will do it in a separate merge
> you might also want to update engines to ^12.20

syavorsky commented:
@brettz9 that seems to be it then. Bumping the minor version seemed to be more appropriate
```
+ comment-parser@1.2.0
```

Thanks for helping to land this!
